### PR TITLE
Comparator materialdata

### PR DESCRIPTION
**Pull request by BehrRiley**
Adds mechanism for adjusting the switch for comparators and a tag for what it's current state is.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialComparator.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    public void setMode(String mode) { getComparator().setMode(Comparator.Mode.valueOf(mode)); }
+
+
+    @Override
+    public String getPropertyString() {
+        return getComparator().getMode().name();
+    }
+
+    @Override
+    public String getPropertyId() { return "mode"; }
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialComparator.java]: What is this formatting?

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialComparator.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        PropertyParser.<MaterialComparator>registerTag("mode", (attribute, material) -> {
+            return new ElementTag(material.getComparator().getMode().name());
+        });
+    }
+
+    public Comparator getComparator() {
+        return (Comparator) material.getModernData().data;
+    }
+
+    public void setMode(String mode) { getComparator().setMode(Comparator.Mode.valueOf(mode)); }
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialComparator.java]: What is this formatting?

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialComparator.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizen.utilities.debugging.Debug;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.properties.Property;
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.block.data.type.Comparator;
+
+public class MaterialComparator implements Property {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialComparator.java]: This should be `MaterialMode`

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialMode.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <MaterialTag.mode>
+        // @returns ElementTag
+        // @mechanism MaterialTag.mode
+        // @group properties
+        // @description
+        // Returns a comparator's mode.
+        // Output is COMPARISON or SUBTRACTION.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialMode.java]: `SUBTRACT` without the `ion`

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java [hunk 5]
```diff
@@ -144,6 +144,7 @@ public static void registermainProperties() {
         PropertyParser.registerProperty(ItemUnbreakable.class, ItemTag.class);
 
         // register core MaterialTag properties
+        PropertyParser.registerProperty(MaterialMode.class, MaterialTag.class);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: This is on very much the wrong line

Comment (BehrRiley) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: Which line does it go on?

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialMode.java (hunk 4 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialMode.java]: and `compare` without the `ison` too

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java (hunk 5 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: Look at the other properties and think about it

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialMode.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        getComparator().setMode(Comparator.Mode.valueOf(mode));
+    }
+
+    @Override
+    public String getPropertyString() {
+        return getComparator().getMode().name();
+    }
+
+    @Override
+    public String getPropertyId() { return "mode"; }
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialMode.java]: Format is borked

**Comment by BehrRiley** [comment]
Noting one of your commits ordering them alphabetically, I put it in the right place and ordered it as well

**Pull request merged**